src/terminal/parser/stateMachine.cpp

```diff
@@ -1270,6 +1448,217 @@ void StateMachine::_EventVt52Param(const wchar_t wch)
     }
 }
 
+// Routine Description:
+// - Processes a character event into an Action that occurs while in the DcsEntry state.
+//   Events in this state will:
+//   1. Ignore C0 control characters
+//   2. Ignore Delete characters
+//   3. Begin to ignore all remaining characters when an invalid character is detected (DcsIgnore)
+//   4. Store parameter data
+//   5. Collect Intermediate characters
+//   6. Pass through everything else
+//  DCS sequences are structurally almost the same as CSI sequences, just with a
+//      extra data string. It's safe to reuse CSI functions for
+//      determining if a character is a parameter, delimiter, or invalid.
+// Arguments:
+// - wch - Character that triggered the event
+// Return Value:
+// - <none>
+void StateMachine::_EventDcsEntry(const wchar_t wch)
+{
+    _trace.TraceOnEvent(L"DcsEntry");
+    if (_isC0Code(wch))
+    {
+        _ActionIgnore();
+    }
+    else if (_isDelete(wch))
+    {
+        _ActionIgnore();
+    }
+    else if (_isCsiInvalid(wch))
+    {
+        _EnterDcsIgnore();
+    }
+    else if (_isNumericParamValue(wch) || _isParameterDelimiter(wch))
+    {
+        _ActionParam(wch);
+        _EnterDcsParam();
+    }
+    else if (_isIntermediate(wch))
+    {
+        _ActionCollect(wch);
+        _EnterDcsIntermediate();
+    }
+    else
+    {
+        _ActionDcsPassThrough(wch);
+        _EnterDcsPassThrough();
+    }
+}
+
+// Routine Description:
+// - Processes a character event into an Action that occurs while in the DcsIgnore state.
+//   Events in this state will:
+//   1. Enter ground on a String terminator
+//   2. Ignore everything else.
+// Arguments:
+// - wch - Character that triggered the event
+// Return Value:
+// - <none>
+void StateMachine::_EventDcsIgnore(const wchar_t wch) noexcept
+{
+    _trace.TraceOnEvent(L"DcsIgnore");
+    if (_isStringTerminator(wch))

```

Comment:
That may have to be another state (DcsIgnoreTerminator) because we'd need to return to DcsIgnore if we got `ESC <anything>` where `anything` was not `\`.

Feel free to correct me if I'm wrong :)

Reply:
Just so I understand, DcsIgnore is "ignore all contents until ST", right?
I know that CsiIgnore is the opposite of this (ignore only specific characters, then drop to ground)

Reply:
**OKAY**, I understand now. When we're in DcsIgnore _esc is handled normally_. We get an `ESC \` and it's handled at ProcessCharacter:1684 because we're not in Osc/DcsPass

Reply:
Yeah James and I had the same dicussion somewhere else. I guess this is OK.